src/allmydata/scripts/tahoe_invite.py

```diff
@@ -0,0 +1,109 @@
+import json
+from os.path import join
+from ConfigParser import NoSectionError
+
+from twisted.python import usage
+from twisted.internet import defer, reactor
+
+from wormhole import wormhole
+
+from allmydata.util import configutil
+from allmydata.util.encodingutil import argv_to_abspath
+from allmydata.scripts.common import get_default_nodedir
+
+
+class InviteOptions(usage.Options):
+    synopsis = "[options] <nickname>"
+    description = "Create a client-only Tahoe-LAFS node (no storage server)."
+
+    optParameters = [
+        ("shares-needed", None, None, "How many shares are needed to reconstruct files from this node"),
+        ("shares-happy", None, None, "Distinct storage servers new node will upload shares to"),
+        ("shares-total", None, None, "Total number of shares new node will upload"),
+    ]
+
+    def parseArgs(self, *args):
+        if len(args) != 1:
+            raise usage.UsageError(
+                "Provide a single argument: the new node's nickname"
+            )
+        self['nick'] = args[0].strip()
+
+
+@defer.inlineCallbacks
+def invite(options):
+    if options.parent['node-directory']:
+        basedir = argv_to_abspath(options.parent['node-directory'])
+    else:
+        basedir = get_default_nodedir()
+    config = configutil.get_config(join(basedir, 'tahoe.cfg'))
+    out = options.stdout
+    err = options.stderr
+
+    try:
+        introducer_furl = config.get('client', 'introducer.furl')
+    except NoSectionError:
+        # we're not a client; maybe this is running *on* the introducer?
+        try:
+            with open(join(basedir, "private", "introducer.furl"), "r") as f:
+                introducer_furl = f.read().strip()
+        except IOError:
+            print >>err, "Can't find introducer fURL in tahoe.cfg nor " + \
+                "{}/private/introducer.furl".format(basedir)
+            raise SystemExit(1)
+
+    nick = options['nick']
+
+    config = {
+        "shares-needed": options["shares-needed"] or config.get('client', 'shares.needed'),
```

Comment:
I suppose `0` is not a very interesting or useful value for any of these configuration options .... if it were I could imagine some surprising results from this code.